starport/templates/typed/indexed/stargate/proto/{{moduleName}}/{{typeName}}.proto.plush

```diff
@@ -0,0 +1,11 @@
+syntax = "proto3";
+package <%= nodash(OwnerName) %>.<%= AppName %>.<%= ModuleName %>;
+
+option go_package = "<%= ModulePath %>/x/<%= ModuleName %>/types";
+
+import "gogoproto/gogo.proto";
+
+message <%= title(TypeName) %> {
+  string index = 2;<%= for (i, field) in Fields { %>
```

Comment:
```suggestion
  string index = 1;<%= for (i, field) in Fields { %>
```

Not blocking.

Reply:
We don't have a `creator` field here anymore, is it fine to not have it?

Reply:
Also, what if instead of creating a new message type, can't we have an option in the existent one that enables indexing with the user given string instead? By default indexing is disabled.

This feels like will simplify our code generation logic and can be more obvious for the user.

Reply:
For now the type is only managed internally, you may have no creator, it's up to the developer to decide. For example, in the IBC exchange, you will have an `OrderBook` which is indexed by `PortID*ChannelId*TokenPair` you don't have creator for this one.

> Also, what if instead of creating a new message type, can't we have an option in the existent one that enables indexing with the user given string instead? By default indexing is disabled.

Indexed type respond to a different logic, methods are a bit different than the other type. And you have less method generated. Also we don't know how those two kind of type will evolve in the future. We need a different template for better maintainability.